apollo-normalized-cache/src/main/java/com/apollographql/apollo/cache/normalized/internal/NoOpApolloStore.kt

```diff
@@ -0,0 +1,133 @@
+import com.apollographql.apollo.cache.normalized.CacheKeyResolver
+import com.apollographql.apollo.cache.normalized.NormalizedCache
+import com.apollographql.apollo.cache.normalized.Record
+import java.util.UUID
+
+/**
+ * An alternative to RealApolloStore for when a no-operation cache is needed.
+ */
+class NoOpApolloStore : ApolloStore, ReadableStore, WriteableStore {
+  override fun merge(recordCollection: Collection<Record>, cacheHeaders: CacheHeaders): Set<String> {
+    return emptySet()
+  }
+
+  override fun merge(record: Record, cacheHeaders: CacheHeaders): Set<String> {
+    return emptySet()
+  }
+
+  override fun read(key: String, cacheHeaders: CacheHeaders): Record? {
+    return null
+  }
+
+  override fun read(keys: Collection<String>, cacheHeaders: CacheHeaders): Collection<Record> {
+    return emptySet()
+  }
+
+  override fun subscribe(subscriber: ApolloStore.RecordChangeSubscriber) {}
+  override fun unsubscribe(subscriber: ApolloStore.RecordChangeSubscriber) {}
+  override fun publish(keys: Set<String>) {}
+  override fun clearAll(): ApolloStoreOperation<Boolean> {
+    return emptyOperation(java.lang.Boolean.FALSE)
+  }
+
+  override fun remove(cacheKey: CacheKey, cascade: Boolean): ApolloStoreOperation<Boolean> {
+    return emptyOperation(java.lang.Boolean.FALSE)
+  }
+
+  override fun remove(cacheKey: CacheKey): ApolloStoreOperation<Boolean> {
+    return emptyOperation(java.lang.Boolean.FALSE)
+  }
+
+  override fun remove(cacheKeys: List<CacheKey>): ApolloStoreOperation<Int> {
+    return emptyOperation(0)
+  }
+
+  override fun networkResponseNormalizer(): ResponseNormalizer<Map<String, Any>> {
+    return ResponseNormalizer.NO_OP_NORMALIZER as ResponseNormalizer<Map<String, Any>>
+  }
+
+  override fun cacheResponseNormalizer(): ResponseNormalizer<Record> {
+    return ResponseNormalizer.NO_OP_NORMALIZER as ResponseNormalizer<Record>
+  }
+
+  override fun <R> readTransaction(transaction: Transaction<ReadableStore, R>): R {
+    return transaction.execute(this)!!
+  }
+
+  override fun <R> writeTransaction(transaction: Transaction<WriteableStore, R>): R {
+    return transaction.execute(this)!!
+  }
+
+  override fun normalizedCache(): NormalizedCache {
+    error("Cannot get normalizedCache: no cache configured")
+  }
+
+  override fun cacheKeyResolver(): CacheKeyResolver {
+    error("Cannot get cacheKeyResolver: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>): ApolloStoreOperation<T> {
+    error("Cannot read operation: no cache configured")
+  }
+
+  override fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>, responseFieldMapper: ResponseFieldMapper<D>,
+      responseNormalizer: ResponseNormalizer<Record>, cacheHeaders: CacheHeaders): ApolloStoreOperation<Response<T>> {
+    error("Cannot read operation: no cache configured")
```

Comment:
Same as above, behaviour change. It used to return emptyOperation(null), which I believe would also crash. The rationale here is that calling these do not make any sense if no cache is configured.

Reply:
Ok, that didn't work. The NO_OP cache is always called, looking into it...

Reply:
Looks like only this method was called. I still made the other calls safer when it was possible.